homeassistant/components/homewizard/config_flow.py

```diff
@@ -106,107 +92,88 @@ async def async_step_zeroconf(
         if (discovery_info.properties[CONF_PATH]) != "/api/v1":
             return self.async_abort(reason="unsupported_api_version")
 
-        # Sets unique ID and aborts if it is already exists
-        await self._async_set_and_check_unique_id(
-            {
-                CONF_IP_ADDRESS: discovery_info.host,
-                CONF_PRODUCT_TYPE: discovery_info.properties[CONF_PRODUCT_TYPE],
-                CONF_SERIAL: discovery_info.properties[CONF_SERIAL],
-            }
+        await self.async_set_unique_id(
+            f"{discovery_info.properties[CONF_PRODUCT_TYPE]}"
+            f"_{discovery_info.properties[CONF_SERIAL]}"
+        )
+        self._abort_if_unique_id_configured(
+            updates={CONF_IP_ADDRESS: discovery_info.host}
         )
 
-        # Pass parameters
-        self.config = {
-            CONF_API_ENABLED: discovery_info.properties[CONF_API_ENABLED],
-            CONF_IP_ADDRESS: discovery_info.host,
-            CONF_PRODUCT_TYPE: discovery_info.properties[CONF_PRODUCT_TYPE],
-            CONF_PRODUCT_NAME: discovery_info.properties[CONF_PRODUCT_NAME],
-            CONF_SERIAL: discovery_info.properties[CONF_SERIAL],
-        }
+        self.discovery = DiscoveryData(
+            ip=discovery_info.host,
+            product_type=discovery_info.properties[CONF_PRODUCT_TYPE],
+            product_name=discovery_info.properties[CONF_PRODUCT_NAME],
+            serial=discovery_info.properties[CONF_SERIAL],
+        )
```

Comment:
Maybe this can be moved a bit up (above line 95) and use `self.discovery` fields to set `unique_id`?